storage/filestore/store.go

```diff
@@ -79,7 +109,45 @@ func (store *Store) Create(ctx context.Context, ref storage.BlobRef, size int64)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
-	return newBlobWriter(ref, store, file), nil
+	return newBlobWriter(ref, store, storage.MaxStorageFormatVersionSupported, file), nil
+}
+
+// SpaceUsed adds up the space used in all namespaces for blob storage
+func (store *Store) SpaceUsed(ctx context.Context) (space int64, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	var totalSpaceUsed int64
+	namespaces, err := store.GetAllNamespaces(ctx)
```

Comment:
Any reason this sin't using `filepath.Walk`?

Reply:
Do you mean GetAllNamespaces() specifically, or this SpaceUsed() function? GetAllNamespaces would not be a great fit, since it only needs to iterate one directory, and this SpaceUsed function offloads the actual directory traversal work to SpaceUsedInNamespace.

Reply:
Just in general. I was wondering whether a top-level regular iteration for namespaces and then using filepath.Walk might end up cleaner than using Readdir at all levels.

Reply:
If it was using Readdir at all levels, then yeah, filepath.Walk would be better. But with a rigid directory hierarchy where each level has a specific meaning, filepath.Walk seems like extra work.